inferno-ml-server-types/src/Inferno/ML/Server/Types.hs

```diff
@@ -600,9 +600,16 @@ data InferenceParam uid gid p s = InferenceParam
     -- For existing inference params, this is the foreign key for the specific
     -- script in the 'InferenceScript' table (i.e. a @VCObjectHash@)
     script :: s,
-    -- | This needs to be linked to a specific version of a model rather
-    -- than the @model@ table itself
-    model :: Id (ModelVersion uid gid Oid),
+    -- | All of the (specific versions of) models that can be used with this
+    -- parameters. @inferno-ml-server@ will copy the contents of each of the
+    -- model versions when evaluating inference scripts. Inference scripts can
+    -- reference any of the linked model versions by referring to the parent
+    -- model\'s name, e.g. @loadModel "name.ts.pt"@
+    --
+    -- Each element represents the ID of a specific model version. However, due
+    -- to limitations in PostgreSQL, there is no referential integrity; i.e.
+    -- the elements are treated as plain integers
```

Comment:
What PostgreSQL limitations prevent referential integrity? AFAICT this looks like a many-to-one relation between modelVersion and inferenceParam which can be modeled as a inferenceParam foreign key on the modelVersion table

Reply:
The reason I phrased it like that is that arrays of foreign keys are in a recent SQL standard but Postgres doesn't support them